cmd/minishift/cmd/start.go

```diff
@@ -215,101 +197,44 @@ func handleProxies() *util.ProxyConfig {
-	err := addOnManager.Apply(addon.GetExecutionContext(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander))
+	err := util.Retry(3, start)
 	if err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprint("Error executing addon commands: ", err))
-	}
-}
-
-// TODO - persistent volume creation should really be fixed upstream, aka 'cluster up'. See https://github.com/openshift/origin/issues/14076 (HF)
-// configurePersistentVolumes makes sure that the default persistent volumes created by 'cluster up' have the right permissions - see https://github.com/minishift/minishift/issues/856
-func configurePersistentVolumes(addOnManager *manager.AddOnManager, sshCommander provision.SSHCommander, ocRunner *oc.OcRunner) error {
-	// don't apply this if anyuid is not enabled
-	anyuid := addOnManager.Get("anyuid")
-	if anyuid == nil || !anyuid.IsEnabled() {
-		return nil
-	}
-
-	fmt.Print("-- Waiting for persistent volumes to be created ... ")
-
-	hostPvDir := viper.GetString(startFlags.HostPvDir.Name)
-
-	var out, err *bytes.Buffer
-
-	// poll the status of the persistent-volume-setup job to determine when the persitent volume creates is completed
-	for {
-		out = new(bytes.Buffer)
-		err = new(bytes.Buffer)
-		exitStatus := ocRunner.Run("get job persistent-volume-setup -n default -o 'jsonpath={ .status.active }'", out, err)
-
-		if exitStatus != 0 || len(err.String()) > 0 {
-			return errors.New("Unable to monitor persistent volume creation")
-		}
-
-		if out.String() != "1" {
-			break
-		}
-
-		time.Sleep(1 * time.Second)
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error starting the VM: %v", err))
 	}
 
-	// verify the job succeeded
-	out = new(bytes.Buffer)
-	err = new(bytes.Buffer)
-	exitStatus := ocRunner.Run("get job persistent-volume-setup -n default -o 'jsonpath={ .status.succeeded }'", out, err)
-
-	if exitStatus != 0 || len(err.String()) > 0 || out.String() != "1" {
-		return errors.New("Persistent volume creation failed")
+	ip, err := hostVm.Driver.GetIP()
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error determining host ip: %v ", err))
 	}
 
-	cmd := fmt.Sprintf("sudo chmod -R 777 %s/pv*", hostPvDir)
-	sshCommander.SSHCommand(cmd)
-
-	// if we have SELinux enabled we need to sort things out there as well
-	// 'cluster up' does this as well, but we do it here as well to have all required actions collected in one
-	// place, instead of relying on some implicit knowledge on what 'cluster up does (HF)
-	cmd = fmt.Sprintf("sudo which chcon; if [ $? -eq 0 ]; then chcon -R -t svirt_sandbox_file_t %s/pv*; fi", hostPvDir)
-	sshCommander.SSHCommand(cmd)
-
-	cmd = fmt.Sprintf("sudo which restorecon; if [ $? -eq 0 ]; then restorecon -R %s/pv*; fi", hostPvDir)
-	sshCommander.SSHCommand(cmd)
-
-	fmt.Println("OK")
-	fmt.Println()
-
-	return nil
+	return hostVm, ip
 }
 
-func automountHostfolders(driver drivers.Driver) {
+func autoMountHostFolders(driver drivers.Driver) {
```

Comment:
so you did want to change it ;-). had this same question when working on the task.